File: Python/0387_first_non_repeat_char/solution.py
def first_non_repeat_char(string):
    char_count = {}
    for char in string:
        if char not in char_count:
            char_count[char] = 1
        else:
            char_count[char] += 1
    
    for key,value in char_count.items():
        if value == 1:
            return key
    return -1

string = 'abcda'
print(first_non_repeat_char(string))

File: JavaScript/0387_first_non_repeat_char/solution.js
function first_non_repeat(string) {
    let char_count = {};
    for (let char of string) {
        if (char in char_count) {
            char_count[char] ++;
        } else {
            char_count[char] = 1;
        }
    }
    for (let char of string) {
        if (char_count[char] == 1) {
            return char;
        }
    }
    return -1;
}

const string = 'abcda';
console.log(first_non_repeat(string))